tools/common/CoreResolver.cs

```diff
@@ -45,7 +47,12 @@ public void Dispose ()
 
 		public AssemblyDefinition Resolve (AssemblyNameReference name)
 		{
-			return Resolve (name, new ReaderParameters { AssemblyResolver = this });
+			var key = name.ToString ();
+			if (!params_cache.TryGetValue (key, out ReaderParameters parameters)) {
+				parameters = new ReaderParameters { AssemblyResolver = this };
+				params_cache [key] = parameters;
```

Comment:
Why does this need to be keyed using the assembly name? Does anything in the `ReaderParameters` change depending on the assembly?

Reply:
I thought of sharing it across assemblies (one for **in memory**, one for **on disk**).
However it's mutable and it's used by code outside our reach (cecil and linker) making any assumption risky.